Makefile

```diff
@@ -176,15 +176,18 @@ uninstall:
 		$(QUIET_UNINSTALL_EACH)tools/uninstall.sh "$(DESTDIR)$(docdir)/tig/$(doc)";)
 
 clean: clean-test clean-coverage
-	$(Q)$(RM) -r $(TARNAME) *.spec tig-*.tar.gz tig-*.tar.gz.sha256 .deps _book node_modules
+	$(Q)$(RM) -r $(TARNAME) tig-*.tar.gz tig-*.tar.gz.sha256 .deps _book node_modules
 	$(Q)$(RM) -r $(compdb_dir) compile_commands.json
 	$(Q)$(RM) $(EXE) $(TOOLS) $(OBJS) core doc/*.xml src/builtin-config.c
 	$(Q)$(RM) $(OBJS:%.o=%.gcda) $(OBJS:%.o=%.gcno)
 
 distclean: clean
 	$(RM) -r doc/manual.html-chunked autom4te.cache
-	$(RM) doc/*.toc $(ALLDOC) aclocal.m4 configure
-	$(RM) config.h config.log config.make config.status config.h.in
+	$(RM) doc/*.toc configure~
+	$(RM) config.h config.log config.make config.status config.h.in~
+
+realclean: distclean
```

Comment:
cc @sbakker

I also thought that `make clean` deletes all generated files, while `make distclean` preserves things like documentation and configure scripts.

In Tig it's the other way round, which is weird. However, Git does the same, so we should first understand why to see if it makes sense for us to deviate.

AFACT this behavior was originally introduced to Git in https://github.com/git/git/commit/922aa807f5fee97fcb2ccd1d1db63b59bdd02128
this is the log message:

	Do not remove distributed configure script

	Before this patch the clean target has removed the
	configure script that comes with Git tar file.
	That made compiling Git for different architectures
	inconvenient.
	This patch excludes configure from the files to be
	deleted by 'make clean' and adds new target 'distclean'
	to preserve old functionality.

(if we stick to a separate target, maybe clean-all is a more familiar/discoverable name)

Reply:
I'm not stuck on the `pristine` name, it's just something I saw occasionally (possibly along with `mrproper` in the Linux kernel source). The [Standard Targets section(https://www.gnu.org/software/make/manual/html_node/Standard-Targets.html#Standard-Targets) in the [GNU make manual](https://www.gnu.org/software/make/manual/) defines `clean` and `distclean`:
  * `clean` - delete files created by the Makefile during the build (compilation) process (`.o`, `.a`, `.so`, etc.), but not configuration files.
  * `distclean` - delete all files that are created by configuring or building the program (so also `config.status`, but not the `configure` script). It should return the state of the directory to that of the unpacked tarball.
 There's also a `maintainer-clean`, but even that one is not supposed to delete the `configure` script. So `veryclean` is probably the best choice here.

Reply:
`veryclean` would be good for me.